packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -5723,6 +5763,185 @@ class _ScribbleFocusableState extends State<_ScribbleFocusable> implements Scrib
   }
 }
 
+class _Scribe extends StatefulWidget {
+  const _Scribe({
+    required this.child,
+    required this.editableKey,
+    required this.enabled,
+    required this.focusNode,
+    required this.selectionControls,
+  });
+
+  final Widget child;
+  final GlobalKey editableKey;
+  final bool enabled;
+  final FocusNode focusNode;
+  final TextSelectionControls? selectionControls;
+
+  @override
+  State<_Scribe> createState() => _ScribeState();
+}
+
+class _ScribeState extends State<_Scribe> {
+  // The handwriting bounds padding of EditText in Android API 34.
+  static const EdgeInsets _handwritingPadding = EdgeInsets.symmetric(
+    horizontal: 10.0,
+    vertical: 40.0,
+  );
+
+  /// Returns a new Rect whose size has changed by the given padding while
+  /// remaining centered.
+  static Rect _pad(Rect rect, EdgeInsets padding) {
+    return Rect.fromLTRB(
+      rect.left - padding.left,
+      rect.top - padding.top,
+      rect.right + padding.right,
+      rect.bottom + padding.bottom,
+    );
+  }
+
+  /// Given a [Rect] in a [RenderBox]'s local coordinate space, returns that
+  /// [Rect] in global coordinates.
+  static Rect _localToGlobalRect(Rect rect, RenderBox renderBox) {
+    return Rect.fromPoints(
+      renderBox.localToGlobal(rect.topLeft),
+      renderBox.localToGlobal(rect.bottomRight),
+    );
+  }
+
+  Rect? _getHandleRect(TextSelectionHandleType type) {
+    if (widget.selectionControls == null) {
+      return null;
+    }
+    // Do not expand the Rect to kMinInteractiveDimension because it will
+    // targeted by a precise pointing device.
+    return widget.selectionControls!.getHandleRect(type, _renderEditable.preferredLineHeight);
+  }
+
+  RenderEditable get _renderEditable => widget.editableKey.currentContext!.findRenderObject()! as RenderEditable;
+
+  Future<void> _handlePointerEvent(PointerEvent event) async {
+    if (event is! PointerDownEvent
+      || event.kind != ui.PointerDeviceKind.stylus
+      || !(await Scribe.isFeatureAvailable() ?? false)) {
+      return;
+    }
+
+    final RenderBox renderBox = widget.editableKey.currentContext!.findRenderObject()! as RenderBox;
+
+    final Rect renderBoxRect = _localToGlobalRect(renderBox.paintBounds, renderBox);
+    final Rect hitRect = _pad(renderBoxRect, _handwritingPadding);
+    if (!hitRect.contains(event.position)) {
+      return;
+    }
+
+    // A stylus event that starts on a selection handle does not start
+    // handwriting, it moves the handle.
```

Comment:
I wish this logic lives in the TextSelectionHandle (or whatever it is called) since we have different handle classes for android and iOS (I think)?